spec/views/user_spec.rb

```diff
@@ -0,0 +1,111 @@
+                  email: 'dante@example.com')
+    end
+
+    before :each do
+      visit users_path
+    end
+
+    it 'shows the username of all users' do
+      expect(page).to have_content('Tom')
+      expect(page).to have_content('Oli')
+      expect(page).to have_content('Dante')
+    end
+
+    it 'See the profile picture for each user' do
+      all_images = page.all('img')
+      expect(all_images.count).to eq(3)
+    end
+
+    it 'See the number of posts each user has written' do
+      expect(page).to have_content('Number of post: 0')
+    end
+
+    it 'When I click on a user, I am redirected to that user\'s show page.' do
+      click_link 'Tom'
+      expect(page).to have_current_path(user_path('1'))
+    end
+  end
+
+  describe 'show' do
+    before :each do
+      visit new_user_session_path
+      fill_in 'Email', with: 'tom@example.com'
+      fill_in 'Password', with: '222555'
+      click_button 'Log in'
+
+      @post1 = Post.find_by(title: 'Post #1')
+      @post2 = Post.find_by(title: 'Post #2')
+      @post3 = Post.find_by(title: 'Post #3')
+
+      if @post1.nil?
+        @post1 = Post.create(author: @first_user, title: 'Post #1', text: 'Post #1', comments_counter: 0,
+                             likes_counter: 0)
+      end
+
+      if @post2.nil?
+        @post2 = Post.create(author: @first_user, title: 'Post #2', text: 'Post #2', comments_counter: 0,
+                             likes_counter: 0)
+      end
+
+      if @post3.nil?
+        @post3 = Post.create(author: @first_user, title: 'Post #3', text: 'Post #3', comments_counter: 0,
+                             likes_counter: 0)
+      end
+    end
+    before :each do
+      visit "/users/#{@first_user.id}"
+    end
+    # let!(:create_post) { Post.create!(post_params) }
+    it 'shows the correct path' do
+      expect(page).to have_current_path(user_path(@first_user.id))
+    end
+
+    it 'shows the user profile picture' do
+      all_images = page.all('img')
+      expect(all_images.count).to eq(1)
+    end
+
+    it 'shows the user username' do
+      expect(page.find('h4', text: 'Tom')).to be_truthy
+    end
+
+    it 'shows the user post count' do
+      expect(page).to have_content("Number of post: #{@first_user.posts.count}")
+    end
+
+    it 'I can see the user\'s bio.' do
+      expect(page).to have_content(@first_user.bio)
+    end
+
+    it 'shows the user\'s first three posts' do
+      expect(page.find_all('h4', class: 'post-title').count).to eq(3)
+    end
+
+    it 'shows the user\'s posts when any post is clicked' do
+      click_link @post1.title
+      expect(page).to have_current_path("/users/#{@post1.author_id}/posts/#{@post1.id}")
+    end
+
+    it 'shows the post details when any post is clicked in the post index page' do
+      click_link @post1.title
+      expect(page).to have_current_path(user_post_path(@first_user, @post1))
+    end
+  end
+end
```

Comment:
- [ ] Good job so far the tests you've covered passed, however, there are some tests that you need to write to complete this milestone.

- [ ] Kindly add a test that redirects to the user's post's index page when you click on a `see all posts`.

- [ ] Kindly write featured tests for `User` **post index page**:
* Tests that asserts you can see the user's profile picture.
* Tests that asserts you can see the user's username.
* Tests that asserts you can see the number of posts the user has written.
* Tests that asserts you can see a post's title.
* Tests that asserts you can see some of the post's body.
* Tests that asserts you can see the first comments on a post.
* Tests that asserts you can see how many comments a post has.
* Tests that asserts you can see how many likes a post has.
* Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
* Tests that redirects to the post's show page when you click on a post.

- [ ] Kindly write featured tests for `User` **post show page**:
* Tests that asserts you can see the post's title.
* Tests that asserts you can see who wrote the post.
* Tests that asserts you can see how many comments it has.
* Tests that asserts you can see how many likes it has.
* Tests that asserts you can see the post body.
* Tests that asserts you can see the username of each commentor.
* Tests that asserts you can see the comment each commentor left.